Fct jl

jakobliebmann opened the pull request:
hi Carl,
hi all,

restored the file forcats.Rmd, restored the intro, proposed a chapter-structure and did some work about general things regarding factors.

Feel free to edit as u like!

Best, Jakob

jakobliebmann committed:
created chapter "forcats" after deletion and inserted intro again

jakobliebmann committed:
Setting a first structure of the chapter and outlining a few general functions

jakobliebmann committed:
continued with some general things

jakobliebmann committed:
adding the last topic of "general": unique values

carlschmidt26 commented on forcats.Rmd:
````diff
@@ -0,0 +1,90 @@
+
+```{r message=FALSE, warning=FALSE, paged.print=TRUE}
+library(tidyverse)
+diamonds<-read_csv("data/diamonds.csv")
````
I would suggest to use spaces to highlight important operators.
E.g. I would change ```diamonds<-read_csv("data/diamonds.csv")``` to  ```diamonds <- read_csv("data/diamonds.csv")```.

The latter tends to be more readable. See also the [Style Guide](http://adv-r.had.co.nz/Style.html) of Advanced R by Hadley Wickham. 

In my opinion the same goes for the pipe operator `%>%` which isn't explicitly mentioned in the guide. 
The pipe operator will also be automatically inserted with blank spaces using the `Ctrl` + `Shift` + `M` shortcut.

jakobliebmann replied:
That's fine...i'll put spaces in the code

carlschmidt26 committed:
Changed a typo in line 68

carlschmidt26 commented on forcats.Rmd:
````diff
@@ -0,0 +1,90 @@
+levels_clarity<-c("I1", "SI2", "SI1", "VS2", "VS1", "VVS2", "VVS1", "IF")
+```
+
+In the next step we lay this newly created levels vector on the column "clarity" of our data set to create a factor. This is a base-R function.
````
Make clear which function is a base-R function.

jakobliebmann replied:
Right, that's misleading. I'll solve this

carlschmidt26 commented on forcats.Rmd:
````diff
@@ -0,0 +1,90 @@
+factor_clarity<-factor(x = diamonds$clarity, levels = levels_clarity)
+```
+
+Another possibility to create a factor is to convert a vector using ***as_factor().***. Here the levels are created automatically. Their order depends on the appearance of the corresponding value in the source vector.
````
Maybe add an example of the order in `diamonds$clarity` before applying `as_factor()`

jakobliebmann replied:
thought I had done that with my first plot....

carlschmidt26 reviewed (commented):
Hi Jakob,

I added some remarks. They are just for you to be aware of these points.
I am happy to work on theses issues if we agree to apply any changes.

Best Carl

jakobliebmann commented:
Thx for your comments Carl! I'll go on it. One thing to think about: is it realy necessary to outline the original order once more?

carlschmidt26 commented:
> Thx for your comments Carl! I'll go on it. One thing to think about: is it realy necessary to outline the original order once more?

The output of the code chunks isn't printed. If the original order is already somewhere close by, it should be fine. Let's come back to that when I've found the time to have a look at what R Studio produces.

jakobliebmann committed:
put space around operators, resolved a misleading expression

jakobliebmann committed:
resolved a merge colflict (accepted Carl's change)

Merge branch 'fct-jl' of https://github.com/jakobliebmann/R-data-science-book into fct-jl

# Conflicts:
#	forcats.Rmd

jakobliebmann committed:
Merge branch 'master' into fct-jl

jakobliebmann commented:
5 minutes ago a merge conflict appeared as till created the file "forcats.Rmd"...i solved this, merging should now be possible after review

carlschmidt26 reviewed (approved):
Document looks nice and neat.
